cmd/av/stack_branchcommit.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"regexp"
+	"strings"
+
+	"emperror.dev/errors"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/meta"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	// This is an arbitrary limit on the branch name length.
+	branchNameLength = 200
+)
+
+var (
+	// See man 1 git-check-ref-format for the refname spec. This pattern is more restrictive
+	// than the spec.
+	//
+	// * Do not allow slashes because creating a branch directory from a commit message is
+	//   unlikely.
+	// * Do not allow dots because dots cannot be placed on a certain location and it's unlikely
+	//   the user wants to have a dot in the branch name.
+	branchNameReplacedPattern = regexp.MustCompile("[^-_a-zA-Z0-9]")
+
+	multipleSpacePattern = regexp.MustCompile(" +")
+)
+
+var stackBranchCommitFlags struct {
+	// The commit message.
+	Message string
+
+	// Name of the new branch.
+	BranchName string
+
+	// Same as `git commit --all`.
+	All bool
+}
+
+var stackBranchCommitCmd = &cobra.Command{
+	Use:          "branch-commit [flags]",
+	Short:        "create a new stacked branch and a commit",
+	Long:         "Create a new branch that is stacked on the current branch, and call git-commit with the specified arguments.",
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		branchName := stackBranchCommitFlags.BranchName
+		if branchName == "" {
+			if stackBranchCommitFlags.Message == "" {
+				_ = cmd.Usage()
+				return errors.New("Need a branch name or a commit message")
```

Comment:
Maybe. Though, (1) entering branch via $EDITOR doesn't feel good to me somehow and (2) if it's for commit message, we will lose the features that `git commit` provide for the commit message editing. We can revisit this if this affects UX.

Reply:
Oh that's fair. I was imagining opening an editor to determine the commit message and then deriving the branch name from that... but that wouldn't let us use vanilla `git commit` since we'd want to check out the new branch first. Oh well, it's not a big deal for now.